dev/ProgressRing/ProgressRing.cpp

```diff
@@ -41,18 +47,98 @@ void ProgressRing::OnRangeBasePropertyChanged(const winrt::DependencyObject&, co
     UpdateSegment();
 }
 
+void ProgressRing::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
+void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    ApplyLottieAnimation();
+}
+
 void ProgressRing::OnStrokeThicknessPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    ApplyLottieAnimation();
     UpdateRing();
 }
 
-winrt::Size ProgressRing::ComputeEllipseSize(double thickness, double actualWidth, double actualHeight)
+void ProgressRing::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressRing::ApplyLottieAnimation()
+{
+    if (auto&& player = m_player.get())
+    {
+        // ProgressRing only accounts for ActualWidth to ensure that it is always a circle
+        const float diameter = static_cast<float>(ActualWidth());
+        const auto size = winrt::Size({ diameter, diameter });
+
+        const auto [foregroundColor, backgroundColor] = [
+            foreground = Foreground().try_as<winrt::SolidColorBrush>(),
+            background = Background().try_as<winrt::SolidColorBrush>()]()
+        {
+            winrt::Color foregroundColor;
+            winrt::Color backgroundColor;
+
+            if (foreground)
+            {
+                foregroundColor = foreground.Color();
+            }
+            else
+            {
+                foregroundColor = SharedHelpers::FindResource(s_DefaultForegroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
+            }
+
+            if (background)
+            {
+                backgroundColor = background.Color();
+            }
+            else
+            {
+                backgroundColor = SharedHelpers::FindResource(s_DefaultBackgroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color();
```

Comment:
>haredHelpers::FindResource(s_DefaultBackgroundThemeResourceName, winrt::Application::Current().Resources()).as<winrt::SolidColorBrush>().Color(); [](start = 35, length = 145)

This crazy long call deserves an abstraction, e.g. GetThemeColor(String resourceName)
[https://en.wikipedia.org/wiki/Don%27t_repeat_yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
+1 I see a handful of call sites that can be easier to read by adding another helper method to SharedHelpers that reads from the application current resources. Something along the lines of SharedHelpers::FindInApplicationResources